File: api-paython/create.py
import json
import os
import logging
import uuid
import time

import boto3


dynamoDb = boto3.resource('dynamodb')


def create(event, context):
    data = json.loads(event['body'])

    if 'nombre' not in data:
        logging.error('Validacion fallida')
        raise Exception("No se creó el elemento")

    timestamp = int(time.time() * 1000)

    table = dynamoDb.Table(os.environ('python_users'))

    item = {
        'id': str(uuid.uuid1()),
        'nombre': data['nombre'],
        'checked': False,
        'creado': timestamp,
        'actualizado': timestamp
    }

    table.put_item(Item=item)

    response = {
        "statusCode": 200,
        "body": json.dumps(item)
    }

    return response

File: employee/employees/employee/create.js
'use strict'

const uuid = require('uuid')
const AWS = require('aws-sdk')

const dynamodb = new AWS.DynamoDB.DocumentClient()

module.exports.create = (event, context, callback) => {

  const time = new Date().getTime()
  const data = JSON.parse(event.body)

  if (typeof data.name !== 'string') {
    console.log('validación error')

    callback(null, {
      statusCode: 400,
      headers: {'Content-Type': 'application/json'},
      body: 'error al crear el elemento'
    })
    return
  }

  const params = {
    TableName: process.env.employees,
    Item: {
      id: uuid.v1(),
      name: data.name,
      age: data.age,
      mail: data.mail,
      checked: false,
      createAt: time,
      updateAt: time
    }
  }

  dynamodb.put(params, (error) => {
    if (error) {
      console.log(error)
      callback(null, {
        statusCode: error.statusCode || 501,
        headers: {'Content-Type': 'application/json'},
        body: 'no se pudo crear el elemento'
      })
      return
    }
    const response = {
      statusCode: 200,
      headers: {'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*'},
      body: JSON.stringify(params.Item)
    }
    callback(null, response)
  })
}
